### ISSUE TITLE
Fetching branches fails for single root implementation

### ISSUE THREAD

**Issue by mwalt** [blt04/doctrine2-nestedset#1]
First of all I would like to thank you for providing this useful extension! I discovered an issue refering to the single root implementation. If you try to fetch a branch by providing a pk and a level, the query which is built automatically by the extension also tries to add the root id (which does not exist) to the where clause of the query so that the query fails (semantic exception).

For example on line 195 in Manager.php:

```
 if($this->getConfiguration()->isRootFieldSupported())
 {
      $qb->andWhere("$alias.$rootField = :rootid")
           ->setParameter('rootid', $node->getRootValue());
 }
```

This happens because the method isRootFieldSupported only checks whether a label for the root field exists in the Config.php class and not whether a corresponding attribute actually exists in the node. You can work around this issues by overwriting the root field name each time you initialize a new NSM:

```
 $config = new Config($em, 'MyNodeClass');
 $config->setRootFieldName(null);
 $nsm = new Manager($config);
```

But I would suggest to use two interfaces for nodes instead. One interface 'INode' for nodes which have just one single root node (getId, get/setLeftValue, get/setRightValue). And in addition, a second interface 'IMultipleRootNode' extending the first one by adding a get/setRootValue method.

The advantages of this solution: no need to add the root value methods to nodes which do not need them (keep code clean) and you can simply check in isRootFieldSupported if the node implements the interface 'IMultipleRootNode' or not.

Comments are welcome!

**Comment by blt04** [comment]
Thanks for the issue report!

This was fixed in SHA: f01665b8dfa73790c4a77b750f73bbf2337629db

Could you please test the [create-root](http://github.com/blt04/doctrine2-nestedset/tree/create-root) branch and reopen if the issue your problem still exists.

**Comment by mwalt** [comment]
Thanks a lot for solving the problem so quickly! I have tested the changes in my concrete setting (single root only) and they seem to work pretty fine!
